src/vs/workbench/contrib/markdown/common/markedKatexExtension.ts

```diff
@@ -32,11 +32,15 @@ export namespace MarkedKatexExtension {
 		return (token: marked.Tokens.Generic) => {
 			let out: string;
 			try {
-				out = katex.renderToString(token.text, {
+				const html = katex.renderToString(token.text, {
 					...options,
 					throwOnError: true,
 					displayMode: token.displayMode,
 				});
+
+				// Wrap in a container with data-latex attribute as a fallback for extracting the original LaTeX source
+				// This ensures we can always retrieve the source even if the annotation element is not present
+				out = `<span class="vscode-katex-container" data-latex="${escapeHtmlAttribute(token.text)}">${html}</span>`;
```

Comment:
Could we extract the class name to a constant string that we can reference in the copy logic too?